No response when configuration is reloaded over http API

When diagnostics configuration is reloaded over http API the server returns empty response. However, it does reload the configuration.

```
~$ curl -XPOST "http://127.0.0.1:8998/reload"
curl: (52) Empty reply from server
```

Also, exception is found in log file
```
ERROR [NanoHttpd Request Processor (#1)] 2017-03-23 13:39:54,981 MetricsCollector.java:102 - Cannot connect to jmx on 127.0.0.1:7199
java.io.IOException: Failed to retrieve RMIServer stub: javax.naming.ServiceUnavailableException [Root exception is java.rmi.ConnectException: Connection refused to host: 127.0.0.1; nested exception is: 
        java.net.ConnectException: Connection refused (Connection refused)]
        at javax.management.remote.rmi.RMIConnector.connect(RMIConnector.java:369) ~[na:1.8.0_121]
        at javax.management.remote.JMXConnectorFactory.connect(JMXConnectorFactory.java:270) ~[na:1.8.0_121]
        at io.smartcat.cassandra.diagnostics.module.metrics.MetricsCollector.connect(MetricsCollector.java:94) ~[cassandra-diagnostics-core-1.3.9-SNPASHOT.jar:na]
        at io.smartcat.cassandra.diagnostics.module.metrics.MetricsModule.<init>(MetricsModule.java:48) [cassandra-diagnostics-core-1.3.9-SNPASHOT.jar:na]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) [na:1.8.0_121]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) [na:1.8.0_121]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) [na:1.8.0_121]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423) [na:1.8.0_121]
        at io.smartcat.cassandra.diagnostics.DiagnosticsProcessor.createModule(DiagnosticsProcessor.java:87) [cassandra-diagnostics-core-1.3.9-SNPASHOT.jar:na]
        at io.smartcat.cassandra.diagnostics.DiagnosticsProcessor.initModules(DiagnosticsProcessor.java:64) [cassandra-diagnostics-core-1.3.9-SNPASHOT.jar:na]
        at io.smartcat.cassandra.diagnostics.DiagnosticsProcessor.<init>(DiagnosticsProcessor.java:44) [cassandra-diagnostics-core-1.3.9-SNPASHOT.jar:na]
        at io.smartcat.cassandra.diagnostics.Diagnostics.reload(Diagnostics.java:165) [cassandra-diagnostics-core-1.3.9-SNPASHOT.jar:na]
        at io.smartcat.cassandra.diagnostics.api.DiagnosticsApiImpl.reload(DiagnosticsApiImpl.java:39) [cassandra-diagnostics-core-1.3.9-SNPASHOT.jar:na]
        at io.smartcat.cassandra.diagnostics.api.HttpHandler.respond(HttpHandler.java:55) [cassandra-diagnostics-core-1.3.9-SNPASHOT.jar:na]
        at io.smartcat.cassandra.diagnostics.api.HttpHandler.serve(HttpHandler.java:41) [cassandra-diagnostics-core-1.3.9-SNPASHOT.jar:na]
        at fi.iki.elonen.NanoHTTPD$HTTPSession.execute(NanoHTTPD.java:945) [cassandra-diagnostics-core-1.3.9-SNPASHOT.jar:na]
        at fi.iki.elonen.NanoHTTPD$ClientHandler.run(NanoHTTPD.java:192) [cassandra-diagnostics-core-1.3.9-SNPASHOT.jar:na]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_121]
Caused by: javax.naming.ServiceUnavailableException: null
        at com.sun.jndi.rmi.registry.RegistryContext.lookup(RegistryContext.java:136) ~[na:1.8.0_121]
        at com.sun.jndi.toolkit.url.GenericURLContext.lookup(GenericURLContext.java:205) ~[na:1.8.0_121]
        at javax.naming.InitialContext.lookup(InitialContext.java:417) ~[na:1.8.0_121]
        at javax.management.remote.rmi.RMIConnector.findRMIServerJNDI(RMIConnector.java:1955) ~[na:1.8.0_121]
        at javax.management.remote.rmi.RMIConnector.findRMIServer(RMIConnector.java:1922) ~[na:1.8.0_121]
        at javax.management.remote.rmi.RMIConnector.connect(RMIConnector.java:287) ~[na:1.8.0_121]
        ... 17 common frames omitted
Caused by: java.rmi.ConnectException: Connection refused to host: 127.0.0.1; nested exception is: 
        java.net.ConnectException: Connection refused (Connection refused)
        at sun.rmi.transport.tcp.TCPEndpoint.newSocket(TCPEndpoint.java:619) ~[na:1.8.0_121]
        at sun.rmi.transport.tcp.TCPChannel.createConnection(TCPChannel.java:216) ~[na:1.8.0_121]
        at sun.rmi.transport.tcp.TCPChannel.newConnection(TCPChannel.java:202) ~[na:1.8.0_121]
        at sun.rmi.server.UnicastRef.newCall(UnicastRef.java:342) ~[na:1.8.0_121]
        at sun.rmi.registry.RegistryImpl_Stub.lookup(Unknown Source) ~[na:1.8.0_121]
        at com.sun.jndi.rmi.registry.RegistryContext.lookup(RegistryContext.java:132) ~[na:1.8.0_121]
        ... 22 common frames omitted
Caused by: java.net.ConnectException: Connection refused (Connection refused)
        at java.net.PlainSocketImpl.socketConnect(Native Method) ~[na:1.8.0_121]
        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350) ~[na:1.8.0_121]
        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206) ~[na:1.8.0_121]
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188) ~[na:1.8.0_121]
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) ~[na:1.8.0_121]
        at java.net.Socket.connect(Socket.java:589) ~[na:1.8.0_121]
        at java.net.Socket.connect(Socket.java:538) ~[na:1.8.0_121]
        at java.net.Socket.<init>(Socket.java:434) ~[na:1.8.0_121]
        at java.net.Socket.<init>(Socket.java:211) ~[na:1.8.0_121]
        at sun.rmi.transport.proxy.RMIDirectSocketFactory.createSocket(RMIDirectSocketFactory.java:40) ~[na:1.8.0_121]
        at sun.rmi.transport.proxy.RMIMasterSocketFactory.createSocket(RMIMasterSocketFactory.java:148) ~[na:1.8.0_121]
        at sun.rmi.transport.tcp.TCPEndpoint.newSocket(TCPEndpoint.java:613) ~[na:1.8.0_121]
        ... 27 common frames omitted
```

This is produced on local three-node (127.0.0.1, 127.0.0.2, 127.0.0.3) cluster created using ccm tool.

Relevant part of cassandra-diagnostics.yml:

```
# controls if HTTP API is turned on. 'true' by default.
httpApiEnabled: true
# specifies the host/address part for listening TCP socket. '127.0.0.1' by default.
httpApiHost: 127.0.0.1
# specifies the port number for the listening TCP socket. '8998' by default.
httpApiPort: 8998
# if API authorization is enabled, API key must be provided through the 'Authorization' header
httpApiAuthEnabled: false
# API access key
httpApiKey: "diagnostics-api-key"

connector:
  jmxHost: 127.0.0.1
  jmxPort: 7100
```

--- a/cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/Diagnostics.java
+++ b/cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/Diagnostics.java
@@ -124,12 +124,6 @@ public class Diagnostics implements QueryReporter {
         } catch (MBeanRegistrationException | InstanceNotFoundException e) {
             logger.error("Unable to unregister DiagnosticsMBean", e);
         }
-
-        if (httpApi != null) {
-            logger.info("Stopping Diagnostics HTTP API");
-            httpApi.closeAllConnections();
-            httpApi = null;
-        }
     }
 
     @Override

--- a/cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/api/HttpHandler.java
+++ b/cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/api/HttpHandler.java
@@ -41,7 +41,7 @@ public class HttpHandler extends NanoHTTPD {
             return respond(session);
         }
 
-        return newFixedLengthResponse(Status.FORBIDDEN, NanoHTTPD.MIME_PLAINTEXT, "Invalid API key");
+        return newFixedLengthResponse(Status.FORBIDDEN, NanoHTTPD.MIME_PLAINTEXT, "Invalid API key\r\n");
     }
 
     private Response respond(IHTTPSession session) {
@@ -54,10 +54,10 @@ public class HttpHandler extends NanoHTTPD {
         } else if (Method.POST.equals(method) && "/reload".equalsIgnoreCase(uri)) {
             diagnosticsApi.reload();
             return newFixedLengthResponse(Status.OK, NanoHTTPD.MIME_PLAINTEXT,
-                    "Configuration reloaded");
+                    "Configuration reloaded\r\n");
         } else {
             return newFixedLengthResponse(Status.NOT_FOUND, NanoHTTPD.MIME_PLAINTEXT,
-                    "Requested URI " + uri + " not found");
+                    "Requested URI " + uri + " not found\r\n");
         }
     }
 

--- a/cassandra-diagnostics-core/src/test/java/io/smartcat/cassandra/diagnostics/DiagnosticsTest.java
+++ b/cassandra-diagnostics-core/src/test/java/io/smartcat/cassandra/diagnostics/DiagnosticsTest.java
@@ -1,0 +1,17 @@
+package io.smartcat.cassandra.diagnostics;
+
+import org.junit.Test;
+
+import io.smartcat.cassandra.diagnostics.config.Configuration;
+
+public class DiagnosticsTest {
+
+    @Test
+    public void test_diagnostics_reload() {
+        Diagnostics diagnostics = new Diagnostics();
+        diagnostics.activate();
+        Configuration configuration = diagnostics.getConfiguration();
+        diagnostics.reload();
+    }
+
+}

